sdk/core/core-rest-pipeline/src/fetchHttpClient.ts

```diff
@@ -0,0 +1,279 @@
+  const headers = buildPipelineHeaders(httpResponse);
+  const response: PipelineResponse = {
+    request,
+    headers,
+    status: httpResponse.status,
+  };
+
+  const bodyStream = isReadableStream(httpResponse.body)
+    ? buildBodyStream(httpResponse.body, request.onDownloadProgress)
+    : httpResponse.body;
+
+  if (
+    // Value of POSITIVE_INFINITY in streamResponseStatusCodes is considered as any status code
+    request.streamResponseStatusCodes?.has(Number.POSITIVE_INFINITY) ||
+    request.streamResponseStatusCodes?.has(response.status)
+  ) {
+    if (request.enableBrowserStreams) {
+      response.browserStreamBody = bodyStream ?? undefined;
+    } else {
+      const responseStream = new Response(bodyStream);
+      response.blobBody = responseStream.blob();
+    }
+  } else {
+    const responseStream = new Response(bodyStream);
+
+    response.bodyAsText = await responseStream.text();
+  }
+
+  return response;
+}
+
+function setupAbortSignal(request: PipelineRequest): {
+  abortController: AbortController;
+  abortControllerCleanup: (() => void) | undefined;
+} {
+  const abortController = new AbortController();
+
+  // Cleanup function
+  let abortControllerCleanup: (() => void) | undefined;
+
+  /**
+   * Attach an abort listener to the request
+   */
+  let abortListener: ((event: any) => void) | undefined;
+  if (request.abortSignal) {
+    if (request.abortSignal.aborted) {
+      throw new AbortError("The operation was aborted.");
+    }
+
+    abortListener = (event: Event) => {
+      if (event.type === "abort") {
+        abortController.abort();
+      }
+    };
+    request.abortSignal.addEventListener("abort", abortListener);
+    abortControllerCleanup = () => {
+      if (abortListener) {
+        request.abortSignal?.removeEventListener("abort", abortListener);
+      }
+    };
+  }
+
+  // If a timeout was passed, call the abort signal once the time elapses
+  if (request.timeout > 0) {
+    setTimeout(() => {
+      abortController.abort();
+    }, request.timeout);
+  }
+
+  return { abortController, abortControllerCleanup };
+}
+
+/**
+ * Gets the specific error
+ */
+function getError(e: RestError, request: PipelineRequest): RestError {
+  if (e && e?.name === "AbortError") {
+    return e;
+  } else {
+    return new RestError(`Error sending request: ${e.message}`, {
+      code: e?.code ?? RestError.REQUEST_SEND_ERROR,
+      request,
+    });
+  }
+}
+
+/**
+ * Converts PipelineRequest headers to Fetch headers
+ */
+function buildFetchHeaders(pipelineHeaders: PipelineHeaders) {
+  const headers = new Headers();
+  for (const [name, value] of pipelineHeaders) {
+    headers.append(name, value);
+  }
+
+  return headers;
+}
+
+function buildPipelineHeaders(httpResponse: Response): PipelineHeaders {
+  const responseHeaders = createHttpHeaders();
+  httpResponse.headers.forEach((value, key) => {
+    responseHeaders.set(key, value);
+  });
```

Comment:
nit: the headers collection is iterable [1], so you can interact with it directly with for/of: 
```suggestion
  for (const [name, value] of httpResponse.headers) {
    responseHeaders.set(name, value);
  }
```

I actually based the shape of our HttpHeaders on the fetch one. :)

[1]: https://developer.mozilla.org/en-US/docs/Web/API/Headers